### PR TITLE
Snow 1001085 add support for default and named args

### PR DESCRIPTION
**Pull request by sfc-gh-pczajka**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Add "default" argument to snowpark schema .yml, which is passed to CREATE FUNCTION/PROCEDURE command. Named arguments and default values already work out of the box.

### CODE REVIEW

File: src/snowflake/cli/plugins/snowpark/common.py [hunk 1]
```diff
@@ -183,8 +182,19 @@ def create_query(
         return "\n".join(query)
 
 
+def _is_signature_type_a_string(sig_type: str) -> bool:
+    return sig_type.lower() in ["string", "varchar"]
```
Comment (sfc-gh-astus) [inline src/snowflake/cli/plugins/snowpark/common.py:186]: int and number don't need thit check?

File: tests_integration/test_data/projects/snowpark_with_default_values/.gitignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.packages/
+.venv/
+app.zip
+__pycache__
```
Comment (sfc-gh-astus) [inline tests_integration/test_data/projects/snowpark_with_default_values/.gitignore:4]: Is this needed?

File: tests_integration/test_data/projects/snowpark_with_default_values/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+snowflake-snowpark-python
+syrupy
+#snowcli  # for local development
```
Comment (sfc-gh-astus) [inline tests_integration/test_data/projects/snowpark_with_default_values/requirements.txt:3]: Only snowflake-snowpark-python should be ok

File: src/snowflake/cli/plugins/snowpark/common.py (hunk 1 above)
Comment (sfc-gh-astus) [reply src/snowflake/cli/plugins/snowpark/common.py:186]: What with boolean?

File: src/snowflake/cli/api/project/schemas/snowpark.py [hunk 4]
```diff
@@ -12,7 +12,7 @@
     Str,
 )
 
-Argument = RelaxedMap({"name": Str(), "type": Str()})
+Argument = RelaxedMap({"name": Str(), "type": Str(), Optional("default"): Str()})
```
Comment (sfc-gh-turbaszek) [inline src/snowflake/cli/api/project/schemas/snowpark.py:15]: Consider adding note to release notes

File: tests_integration/test_data/projects/snowpark_with_default_values/.gitignore (hunk 2 above)
Comment (sfc-gh-pczajka) [reply tests_integration/test_data/projects/snowpark_with_default_values/.gitignore]: Fixed

File: tests_integration/test_data/projects/snowpark_with_default_values/requirements.txt (hunk 3 above)
Comment (sfc-gh-pczajka) [reply tests_integration/test_data/projects/snowpark_with_default_values/requirements.txt]: fixed

File: src/snowflake/cli/api/project/schemas/snowpark.py (hunk 4 above)
Comment (sfc-gh-pczajka) [reply src/snowflake/cli/api/project/schemas/snowpark.py:15]: I was thinking about that; done :)

File: src/snowflake/cli/plugins/snowpark/common.py (hunk 1 above)
Comment (sfc-gh-pczajka) [reply src/snowflake/cli/plugins/snowpark/common.py:169]: Works out of the box, I've added a test for that

File: tests_integration/test_data/projects/snowpark_with_default_values/snowflake.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+          default: "true"
+        - name: "b2"
+          type: "boolean"
+          default: "True"
+        - name: "f"
+          type: "float"
+          default: "1.5"
+        - name: "l"
+          type: "array"
+          default: "[1, 2, 3]"
```
Comment (sfc-gh-astus) [inline tests_integration/test_data/projects/snowpark_with_default_values/snowflake.yml:40]: 🥇

File: src/snowflake/cli/plugins/snowpark/commands.py [hunk 6]
```diff
@@ -223,8 +224,13 @@ def _deploy_single_object(
     packages: List[str],
     stage_artifact_path: str,
 ):
-    identifier = build_udf_sproc_identifier(object_definition)
-    log.info("Deploying %s: %s", object_type, identifier)
+    identifier = build_udf_sproc_identifier(
+        object_definition, include_parameter_names=False
+    )
```
Comment (sfc-gh-astus) [inline src/snowflake/cli/plugins/snowpark/commands.py:229]: Can be removed

File: RELEASE-NOTES.md [hunk 7]
```diff
@@ -22,6 +22,8 @@
   * Previously created procedures or functions won't work with `deploy` command due to change in stage path of artifact. Previous code will remain under old path on stage.
   * Package commands are now under `snow snowpark package`.
   * Coverage commands were removed. To measure coverage of your procedures or functions use coverage locally.
+  * Add `default` field to procedure and function arguments definition in `snowflake.yml` to support [named and optional
+  arguments](https://docs.snowflake.com/en/developer-guide/udf/udf-calling-sql#calling-a-udf-that-has-optional-arguments)
```
Comment (sfc-gh-turbaszek) [inline RELEASE-NOTES.md:26]: Let's remember to add this to documentation

Comment (sfc-gh-pczajka) [reply RELEASE-NOTES.md:26]: I've reached out to Mike on Slack and in github comments (updating google doc accordingly)

**Review by sfc-gh-astus — DISMISSED** [review]

File: RELEASE-NOTES.md [hunk 8]
```diff
@@ -1,3 +1,8 @@
+# v2.0.1
+
```
Comment (sfc-gh-astus) [inline RELEASE-NOTES.md:2]: `## New additions`

File: RELEASE-NOTES.md [hunk 9]
```diff
@@ -1,5 +1,7 @@
 # v2.0.1
 
+## New additions
+
```
Comment (sfc-gh-astus) [inline RELEASE-NOTES.md:4]: New line is not needed

File: RELEASE-NOTES.md [hunk 10]
```diff
@@ -1,5 +1,7 @@
...
   * Snowpark: add `default` field to procedure and function arguments definition in `snowflake.yml` to support [named and optional
   arguments](https://docs.snowflake.com/en/developer-guide/udf/udf-calling-sql#calling-a-udf-that-has-optional-arguments)
```
Comment (sfc-gh-astus) [inline RELEASE-NOTES.md:6]: Don't have to be indent

File: RELEASE-NOTES.md (hunk 9 above)
Comment (sfc-gh-pczajka) [reply RELEASE-NOTES.md]: done

File: RELEASE-NOTES.md (hunk 10 above)
Comment (sfc-gh-pczajka) [reply RELEASE-NOTES.md]: and done

**Review by sfc-gh-astus — APPROVED** [review]

**Pull request merged**